packages/amplify-category-function/src/provider-utils/awscloudformation/utils/storeResources.ts

```diff
@@ -202,11 +202,9 @@ const layerParamsToStoredParams = (parameters: LayerParameters | StoredLayerPara
 });
 
 function createParametersFile(context, parameters, resourceName, parametersFileName) {
-  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
-  const resourceDirPath = path.join(projectBackendDirPath, categoryName, resourceName);
-  fs.ensureDirSync(resourceDirPath);
-  const parametersFilePath = path.join(resourceDirPath, parametersFileName);
-  const currentParameters = fs.existsSync(parametersFilePath) ? context.amplify.readJsonFile(parametersFilePath) : {};
+  const parametersFilePath = path.join(context.amplify.pathManager.getBackendDirPath(), categoryName, resourceName, parametersFileName);
+  const currentParameters = context.amplify.readJsonFile(parametersFilePath, undefined, false) || {};
+  delete currentParameters.mutableParametersState; // this field was written in error in a previous version of the cli
```

Comment:
I think it might make sense to import `loadFunctionParameters` and use it here

Reply:
lol yeah that would definitely be better.

I'm not 100% sure we even need to read the existing params here because that's how the parameters get initialized in the first place, but I'll load it here as well just to be safe

Reply:
Oh, actually I can't do that here because this method is used for writing files besides the function-parameters file. So I'm going to leave it as is for now